src/lib/util/timers.ml

```diff
@@ -149,6 +149,96 @@ let string_of_ty_function f = match f with
   | F_apply_subst   -> "apply_subst"
   | F_instantiate   -> "instantiate"
 
+module TimerTable = struct
+  module FTbl = Hashtbl.Make(
+    struct
+      type t = ty_function
+
+      let hash = function
+        | F_add           -> 0
+        | F_add_lemma     -> 1
+        | F_assume        -> 2
+        | F_class_of      -> 3
+        | F_leaves        -> 4
+        | F_make          -> 5
+        | F_m_lemmas      -> 6
+        | F_m_predicates  -> 7
+        | F_query         -> 8
+        | F_solve         -> 9
+        | F_subst         -> 10
+        | F_union         -> 11
+        | F_unsat         -> 12
+        | F_add_predicate -> 13
+        | F_add_terms     -> 14
+        | F_are_equal     -> 15
+        | F_none          -> 16
+        | F_new_facts     -> 17
+        | F_apply_subst   -> 18
+        | F_instantiate   -> 19
```

Comment:
We may use the function `Hashtbl.hash`here? I know using this function involves some risks but for a profiling module, the risk is limited.

Reply:
What are the risks of `Hashtbl.hash` you have in mind?

(To be fair I am not sure why we have an explicit datatype to describe profiled functions)

Reply:
I mean the `hash` function of `Hashtbl` doesn't always produce a good hash function to get a `O(1)` lookup function in your hash table but in our case, it doesn't matter as the hash table is tiny.

Reply:
This function already was defined, I just copied it.

Reply:
If we want a shorter version, we can use `Obj.magic` which would here work like a charm...  :smiling_imp:

Reply:
Final choice: ppx_deriving.enum, as it defines the values we want.